### ISSUE TITLE
My location marker is rendered badly/pixeled

### ISSUE THREAD

**Issue by boldtrn** [mapsforge/vtm#207]
I played around with vtm on my [Moto G3](http://www.gsmarena.com/motorola_moto_g_(3rd_gen)-7247.php) with an Adreno 306 GPU. The "my location" is is rendered incorrect on that device, see the attached Screenshot. The rest of the map looks ok to me (as far as I can tell).

![my_location](https://cloud.githubusercontent.com/assets/1553525/19483170/bb47e1cc-9553-11e6-94c2-fc2ca75c7273.png)

**Comment by devemux86** [comment]
We have another [report](https://groups.google.com/d/msg/mapsforge-dev/aCTj7IkwqOc/ZsgCJlE2EwAJ) about the [Motorola Moto G4 Play](http://www.gsmarena.com/motorola_moto_g4_play-8104.php) with Adreno 306 GPU too.

**Comment by ocsike** [comment]
On LG Fino I have same issue  with using  MapFileTileSource or BitmapTileSource.

**Comment by devemux86** [comment]
[LG L Fino](http://www.gsmarena.com/lg_l_fino-6592.php) has an Adreno 302 GPU.

**Comment by ocsike** [comment]
I found older version of vtm here: http://webcache.googleusercontent.com/search?q=cache:wbdNyiIh2VUJ:www.opensciencemap.org/+&cd=2&hl=hu&ct=clnk&gl=hu&client=firefox-b
I downloaded OpenScienceMap 0.5 apk and tried. Result is here (no flickering) :
![2016_10_20_10 14 21](https://cloud.githubusercontent.com/assets/13695075/19552395/f15105d4-96af-11e6-8e02-fe5a5f829c48.png)

**Comment by devemux86** [comment]
What do you mean with flickering?
This issue is about location marker rendering artifacts.

Flickering is currently discussed [here](https://groups.google.com/forum/#!topic/mapsforge-dev/aCTj7IkwqOc) in the forum, where there are some videos too.

**Comment by ocsike** [comment]
Sorry, I wrote flickering instead of rendering artifacts. My mistake.

**Comment by devemux86** [comment]
It's ok, just wanted to clarify the issue. :slightly_smiling_face: 

So it appears that the problem started when we integrated location layer in VTM fork?
The old VTM repository's location layer renders fine at the same device?

**Comment by ocsike** [comment]
Yes, the old version is ok on my device.  The OpenScienceMap 0.5 apk is produced in 2013.

**Comment by devemux86** [comment]
Hmm looking at 0.5 apk native libs, there were some changes since then.
We integrated `vtm-app` and location renderer from latest https://github.com/opensciencemap/vtm-app.
If you could test that and see possible rendering artifacts?

**Comment by ocsike** [comment]
Maybe I found the solution. :)  I changed the floating precision from high to medium in shader code.
I changed the  following line 
### "precision highp float;"  to  "precision mediump float;"  
in vShaderStr,fShaderStr1,fShaderStr2 variables in LocationRenderer.java
And I get  :
![2016_10_20_18 56 21](https://cloud.githubusercontent.com/assets/13695075/19569766/35a6fa1c-96f7-11e6-8109-ac257c92fdff.png)

**Comment by devemux86** [comment]
Oh, I had changed those with 0c898bf6820ab8fc33968b5a38111a659754f834 in #171, following the precision increase of other map shaders (https://github.com/opensciencemap/vtm/issues/148).

Let me see if I can run also a test with the change.
If it's the fix you should post a pull request (as you found it :slightly_smiling_face:) and I could insert another RC in upcoming VTM release!

BTW @hjanetzek any hints?

**Comment by boldtrn** [comment]
> "precision highp float;" to "precision mediump float;"

I can confirm this fix. The my location marker looks god when I change the highhp to mediump.

**Comment by devemux86** [comment]
@boldtrn thanks for the feedback!

@ocsike do you want to post a pull request with the fix?

**Comment by ocsike** [comment]
Yes, post.
